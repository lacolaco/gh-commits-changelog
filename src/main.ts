const mergeConfig = require("conventional-changelog-core/lib/merge-config");
const angular = require("conventional-changelog-angular");
import { createParser } from "./parser";
import { createWriter } from "./writer";
import { GitHubCommit } from "./types";

export type Context = {
  version?: string;
  previousTag?: string;
  currentTag?: string;
  host?: string;
  owner?: string;
  repository?: string;
  repoUrl?: string;
  title?: string;
  date?: string;
};

export async function main(ghCommits: Array<GitHubCommit>, context: Context) {
  const config = await mergeConfig(
    {
      config: angular
    },
    {
      ...context
    }
  );
  const parser = createParser(config.parserOpts);
  const writer = createWriter(config.writerOpts);
  const changelog = writer(parser(ghCommits), config.context);
  return await changelog;
}
