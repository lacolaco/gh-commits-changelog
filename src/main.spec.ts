import { main, Context } from "./main";
import { PullsListCommitsResponse } from "@octokit/rest";

describe("main", () => {
  describe("integration tests", () => {
    let context: Context;
    beforeEach(() => {
      context = {
        host: "https://github.com",
        owner: "angular",
        repository: "angular",
        currentTag: "e37f23f52c0d46126c3d15f6dac2e573c838a353",
        previousTag: "c69fff15c9e001bd636b180758f1102d338564fb",
        version: "Release 2018-11-20",
        date: "2018-11-11"
      };
    });
    test("Work well with Angular commits", async () => {
      const fixture: PullsListCommitsResponse = require("../test/fixtures/angularPullRequestCommits.json");
      const parsed = await main(fixture, context);
      expect(parsed).toMatchSnapshot();
    });

    test("Work well with plain commits", async () => {
      const fixture: PullsListCommitsResponse = require("../test/fixtures/pullRequestCommits.json");
      const parsed = await main(fixture, context);
      expect(parsed).toMatchSnapshot();
    });
  });
});
