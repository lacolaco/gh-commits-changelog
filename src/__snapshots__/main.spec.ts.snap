// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main integration tests Work well with Angular commits 1`] = `
"# [Release 2018-11-20](https://github.com/angular/angular/compare/c69fff15c9e001bd636b180758f1102d338564fb...e37f23f52c0d46126c3d15f6dac2e573c838a353) (2018-11-11)


### Features

* **common:** new HttpClient API ([e37f23f](https://github.com/angular/angular/commit/e37f23f))



"
`;

exports[`main integration tests Work well with plain commits 1`] = `
"# [Release 2018-11-20](https://github.com/angular/angular/compare/c69fff15c9e001bd636b180758f1102d338564fb...e37f23f52c0d46126c3d15f6dac2e573c838a353) (2018-11-11)



"
`;
